jlm/hls/ir/hls.hpp

```diff
@@ -1087,45 +1094,25 @@ class mem_req_op final : public jlm::rvsdg::simple_op
 {
 public:
   virtual ~mem_req_op()
-  {
-    delete LoadTypes_;
-    delete StoreTypes_;
-  }
+  {}
```

Comment:
`~mem_req_op() override = default;`?

Reply:
yes, and generally definitions for any virtual methods should not be in header -- it slows down compilation & linking (and causes version coupling issues for shared libraries, but that's a bit premature to be concerned about this).